### PR TITLE
Implement progress static visualization renderer

### PR DESCRIPTION
**Pull request by alxnddr**
Epic: https://github.com/metabase/metabase/issues/18676

Add progress static viz renderer:
Screen Shot 2021-11-08 at 02 09 26

### How to verify
- Run Metabase locally and go to http://localhost:3000/_internal/static-viz
- Ensure the progress bar static visualization renders correctly

### CODE REVIEW

**Comment by alxnddr** [comment]
@metabase-bot run visual tests

File: frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  borderRadius: 5,
+  labelsMargin: 16,
+  iconSize: 20,
+  pointer: {
+    width: 20,
+    height: 10,
+  },
+};
+
+interface ProgressBarProps {
```
Comment (ranquild) [inline frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx:30]: AFAIK the BE interface for static viz is `{ data, labels, settings }`. Have you confirmed the API with a BE dev?

**Review by ranquild — CHANGES_REQUESTED** [review]
Looks good. The only case that should be implemented is negative `value`, e.g. the product vs static viz:
Screen Shot 2021-11-08 at 1 22 27 pm

Screen Shot 2021-11-08 at 1 23 07 pm

File: frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (howonlee) [reply frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx:30]: yeah, we can futz w/ arity in the BE if you want, that seems awkward as is

**Review by kdoh — COMMENTED** [review]
Looking good to me product wise! Really appreciate you including the different states in _internal/static-viz. Will want to do another review when the BE renderer gets hooked up and I 2nd the comment @ranquild made about negatives but overall looks just like I'd expect.

File: frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/static-viz/components/ProgressBar/ProgressBar.tsx:35]: Fixed that, thanks

**Review by ranquild — APPROVED** [review]

**Pull request merged**